src/Sulu/Bundle/MediaBundle/Media/ImageConverter/ImagineImageConverter.php

```diff
@@ -93,9 +93,12 @@ public function getSupportedOutputImageFormats(?string $mimeType): array
 
     public function convert(FileVersion $fileVersion, $formatKey, $imageFormat)
     {
+        $rawResource = $this->storage->load($fileVersion->getStorageOptions());
+        /** @var string $mimeType */
+        $mimeType = \mime_content_type($rawResource);
         $imageResource = $this->mediaImageExtractor->extract(
-            $this->storage->load($fileVersion->getStorageOptions()),
-            $fileVersion->getMimeType()
+            $rawResource,
+            $fileVersion->getMimeType() ?? $mimeType,
```

Comment:
this can really end in hard errors, resources here arenot able to call `mime_content_type` always as they are in a few scenerries streams. Also a FileVersion without a mimetype should never exist.

So this change should be reverted.

```suggestion
$imageResource = $this->mediaImageExtractor->extract(
            $this->storage->load($fileVersion->getStorageOptions()),
            $fileVersion->getMimeType(),
```

Reply:
Reverted, but well the return type of the function is nullable. So I assumed that this is not 100% guaranteed.